tests/conftest.py

```diff
@@ -315,3 +317,39 @@ async def mqtt_mock(hass, mqtt_client_mock, mqtt_config):
     component = hass.data["mqtt"]
     component.reset_mock()
     return component
+
+
+@pytest.fixture
+async def patchable_time():
+    """Allow time to be patchable by using event listeners instead of asyncio loop."""
+
+    @ha.callback
+    @loader.bind_hass
+    def async_track_point_in_utc_time(hass, action, point_in_time):
+        """Add a listener that fires once after a specific point in UTC time."""
+        # Ensure point_in_time is UTC
+        point_in_time = event.dt_util.as_utc(point_in_time)
+
+        @ha.callback
+        def point_in_time_listener(event):
+            """Listen for matching time_changed events."""
+            now = event.data[ATTR_NOW]
+
+            if now < point_in_time or hasattr(point_in_time_listener, "run"):
+                return
+
+            # Set variable so that we will never run twice.
+            # Because the event bus might have to wait till a thread comes
+            # available to execute this listener it might occur that the
+            # listener gets lined up twice to be executed. This will make
+            # sure the second time it does nothing.
+            setattr(point_in_time_listener, "run", True)
+            async_unsub()
+
+            hass.async_run_job(action, now)
+
+        async_unsub = hass.bus.async_listen(EVENT_TIME_CHANGED, point_in_time_listener)
+
+        return async_unsub
+
+    event.async_track_point_in_utc_time = async_track_point_in_utc_time
```

Comment:
Won't this patch make the tests that use this patch pointless, since they are no longer testing the code under test.

Reply:
This patch restores the old code, which we know also works and has been tested. We can write a test for it to make sure it behaves the same.

There are separate tests to test the new functionality. I agree that it is not as clean.

Reply:
Maybe prefix the fixture with `legacy_` so it is not used in new code, then we can slowly remove it over time.

Reply:
Realized my fixture was overriding the actual method for all tests. Working on a new version 😆